macrel/AMP_features.py

```diff
@@ -2,109 +2,40 @@
 # _*_coding:utf-8_*_
 
 import sys
+import gzip
+import lzma
 import pandas as pd
 import numpy as np
-import rpy2
-import rpy2.robjects
-from rpy2.robjects import numpy2ri
-from .fasta import fasta_iter
-numpy2ri.activate()
-r = rpy2.robjects.r
-r.library('Peptides')
 
-GROUPS_SA = ['ALFCGIVW', 'RKQEND', 'MSPTHY'] #solventaccess
-GROUPS_HB = ['ILVWAMGT', 'FYSQCN', 'PHKEDR'] # HEIJNE&BLOMBERG1979
-
-
-#' # http://emboss.bioinformatics.nl/cgi-bin/emboss/pepstats
-#' # Property      Residues              Number  Mole%
-#' # Tiny          (A+C+G+S+T)             4   19.048
-#' # Small         (A+B+C+D+G+N+P+S+T+V)   4   19.048
-#' # Aliphatic     (A+I+L+V)               5   23.810
-#' # Aromatic      (F+H+W+Y)               5   23.810
-#' # Non-polar     (A+C+F+G+I+L+M+P+V+W+Y) 11  52.381
-#' # Polar         (D+E+H+K+N+Q+R+S+T+Z)   9   42.857
-#' # Charged       (B+D+E+H+K+R+Z)         8   38.095
-#' # Basic         (H+K+R)                 8   38.095
-#' # Acidic        (B+D+E+Z)               0   00.000
-
-_aa_groups = [
-    set('ACGST'),          # Tiny      
-    set('ABCDGNPSTV'),     # Small     
-    set('AILV'),           # Aliphatic 
-    set('FHWY'),           # Aromatic  
-    set('ACFGILMPVWY'),    # Nonpolar  
-    set('DEHKNQRSTZ'),     # Polar     
-    set('BDEHKRZ'),        # Charged   
-    set('HKR'),            # Basic     
-    set('BDEZ'),           # Acidic    
-]
-
-def amino_acid_composition(seq):
-    '''amino_acid_composition: return AA composition fractions'''
-    # See groups above
-    return np.array(
-            [sum(map(g.__contains__, seq)) for g in _aa_groups],
-            dtype=float)/len(seq)
-
-def ctdd(sequence, groups):
-    code = []
-    for group in groups:
-        for i, aa in enumerate(sequence):
-            if aa in group:
-                code.append((i + 1)/len(sequence) * 100)
-                break
-        else:
-            code.append(0)
-    return np.array(code)
+from Bio import SeqIO
+from .database import GROUPS_SA, GROUPS_HB
+from .macrel_features import MacrelFeatures
 
+def openfile(ifile):
```

Comment:
We already had a FASTA parsing function that didn't require Biopython (which is also a heavy dependency for just the FASTA parser)

Reply:
I thought the same, but after the CLI testing, I checked that with the dependencies, Macrel package now is 200MB less heavy than before.